geonode/layers/utils.py

```diff
@@ -788,6 +790,19 @@ def create_thumbnail(instance, thumbnail_remote_url, thumbnail_create_url=None,
 
         if valid_x and valid_y:
             logger.debug('Valid bbox. Will create Remote Thumbnail Link.')
+
+            if check_ogc_backend(qgis_server.BACKEND_PACKAGE):
+                # Only for QGIS Server
+                # We don't want to add the internal QGIS IP.
+                # We need the public URL of Geonode.
+                public_url = settings.SITEURL
+                if public_url.endswith('/'):
+                    # reverse will give also a slash. We need only one.
+                    public_url = public_url[0:-1]
+                public_url += reverse('qgis_server:request')
+                thumbnail_remote_url = thumbnail_remote_url.replace(
+                    QGIS_SERVER_CONFIG['qgis_server_url'], public_url)
```

Comment:
Yes, I didn't realize there are 2 kinds of thumbnail. So, it's best if we fix it in thumbnail generated code, because AFAIK, we don't want to expose QGIS Server to outside.